Junimatic/WorkFinder.cs

```diff
@@ -111,16 +111,23 @@ private void DoJunimos(bool isAutomationInterval)
             }
 
             // Junimos only work on the farm or in farm buildings.
-            var allJunimoFriendlyLocations =
-                Game1.getFarm().buildings
+            var allJunimoFriendlyLocations = new List<GameLocation>();
+            if (ModEntry.Config.AllowAnyLocation)
+            {
+                allJunimoFriendlyLocations = new List<GameLocation>(Game1.game1._locations);
+            }
+            else
+            {
+                allJunimoFriendlyLocations = Game1.getFarm().buildings
                     .Select(b => b.indoors.Value)
                     .Where(l => l is not null).Select(l => l!)
                     .ToList();
-            allJunimoFriendlyLocations.Add(Game1.getFarm());
-            allJunimoFriendlyLocations.AddRange(
-                new string[] { "FarmCave", "IslandWest", "Cellar", "FarmHouse", "IslandFarmHouse" }
-                .Select(name => Game1.getLocationFromName(name))
-                .Where(l => l is not null));
+                allJunimoFriendlyLocations.Add(Game1.getFarm());
+                allJunimoFriendlyLocations.AddRange(
+                    new string[] { "FarmCave", "IslandWest", "Cellar", "FarmHouse", "IslandFarmHouse" }
```

Comment:
Perhaps instead of having the setting Extend this list - maybe this list should be the default value of the setting?

Reply:
That's definitely a cleaner way to do it, I just didn't think of that.